.github/workflows/main.yml

```diff
@@ -419,6 +419,13 @@ jobs:
           mkdir gen/artifacts
           mv gen/Release/*.pkg gen/artifacts
 
+      - name: Save vcpkg cache
+        uses: actions/cache/save@v4
+        if: steps.restore-vcpkg-cache.outputs.cache-hit != 'true'
```

Comment:
This condition does not seems correct for me.
Just me vcpkg do its job. Just use `actions/cache@v4` only and let it restore and save the cache.

Reply:
GitHub Action caches are write once. If you remove this condition, you don't need to write it again *under a different key*. The alternative, which I had in an earlier iteration, was to just use the run id as part of the cache key, but that has the downside that each run will write its own 100 MiB cache, even if nothing changed.

Reply:
Actually, that might only apply to the cache action, and not the cache/save action. I'll verify my assumption.

Reply:
In other words, if you want to have any chance of replacing the cache when something changes, you need to use a key that changes when something important changes (or have some additional code/action to detect cache changes, delete the existing cache and the upload the modified cache under the same key).